### DISCUSSION TITLE
Config.py API Documentation

### DISCUSSION THREAD

**Discussion by MaxWinsemius** [qutebrowser/qutebrowser#7461] [Q&A] [answered]
I love to try and play around with the config.py environment, but I seem to be unable to find any documentation on the file. I've been stuck on it for some hours now, and the only thing I seem to find is reverse engineering the application. I know how to play around with the `:set` command (although even for this I am puzzled on what _all_ the available options are with regard of processing the data). I understand that there are userscripts available, but how would the config file then be interesting to use?

Is there some documentation like [the python sockets api docs](https://docs.python.org/3/library/socket.html) available for qutebrowser?

**Comment by The-Compiler** [comment, marked as answer]
It's not really intended as a Python API (see #30 for that), rather as a slightly more flexible alternative to a purely declarative config format.

The relevant docs are here:

- [Configuring qutebrowser](https://qutebrowser.org/doc/help/configuring.html)
- [Settings reference](https://qutebrowser.org/doc/help/settings.html)

...or all accessible via `:help` in qutebrowser too.
